### PR TITLE
Cbo 45 risk based bills filter

### PR DESCRIPTION
**Pull request by kmahern**
#### What
Make changes to Risk based bills filter

#### Ticket

CBO-45 https://dsdmoj.atlassian.net/browse/CBO-45

#### Why

To enable users to carry out risk based billing

#### How

Made changes to filter classes and associated spec classes to implement functionality described in ticket.

### CODE REVIEW

File: app/interfaces/api/v2/query_helper.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (jsugarman) [inline app/interfaces/api/v2/query_helper.rb]: style wise best not to leave blank lines at top of page

File: app/interfaces/api/entities/search_result.rb [hunk 2]
```diff
@@ -109,7 +109,8 @@ def interim_disbursements
       end
 
       def risk_based_bills
-        ((risk_based_class_letter && contains_risk_based_fee).eql?(true) && is_submitted?).to_i
+        ((risk_based_class_letter && (contains_risk_based_fee ||
```
Comment (jsugarman) [inline app/interfaces/api/entities/search_result.rb]: I'd refactor to use a couple of methods or more with this method being the parent. just coz all the conditional logic and braces start to hurt peoples heads

Comment (kmahern) [reply app/interfaces/api/entities/search_result.rb]: Thanks Joel, will do.

File: spec/models/claims/allocation_filters_spec.rb [hunk 3]
```diff
@@ -36,24 +36,24 @@ module Claims
...
+        # create :interim_claim
+        # create :litigator_claim, offence: offence_a
+        # claim = create :litigator_claim, offence: offence_f
+        # create :graduated_fee, fee_type: guilty_plea_fee_type, claim: claim
+
+        # expect(Claim::BaseClaim.all_risk_based_bills).to match_array( [ claim ])
+      # end
+    # end
   end
 end
```
Comment (jsugarman) [inline spec/models/claims/allocation_filters_spec.rb]: styling - add linebreaks at the end of files

File: app/interfaces/api/entities/search_result.rb (hunk 2 above)
Comment (jsugarman) [inline app/interfaces/api/entities/search_result.rb]: I'd refactor to use methods to isolate the conditional logic here

File: app/interfaces/api/v2/query_helper.rb (hunk 1 above)
Comment (jsugarman) [inline app/interfaces/api/v2/query_helper.rb]: styling - should remove linebreaks from top of files

File: app/models/claims/allocation_filters.rb [hunk 4]
```diff
@@ -45,13 +45,13 @@ def all_graduated_fees
...
     end
 
-    def all_risk_based_bills
-      where(type: 'Claim::LitigatorClaim')
-        .where(offence_id: Offence.joins(:offence_class).where(offence_classes: { class_letter: %w[E F H I] }))
-        .joins(:fees)
-        .where('"fees"."fee_type_id" = ?', Fee::GraduatedFeeType.where(description: 'Guilty plea').pluck(:id).first)
-        .where('"fees"."quantity" between 1 and 50')
-    end
+    # def all_risk_based_bills
```
Comment (jsugarman) [inline app/models/claims/allocation_filters.rb]: could this not be amended to do what you need. I don't have a clear idea of everything going on but be wary of N+1 queries resulting from looping over thousands of records and issuing other queries for each one. I appreciate this is a tricky query though.

if this isn't used/required then should delete

File: spec/models/claims/allocation_filters_spec.rb [hunk 5]
```diff
@@ -36,24 +36,24 @@ module Claims
...
-        create :transfer_claim
-        create :interim_claim
-        create :litigator_claim, offence: offence_a
-        claim = create :litigator_claim, offence: offence_f
-        create :graduated_fee, fee_type: guilty_plea_fee_type, claim: claim
-
-        expect(Claim::BaseClaim.all_risk_based_bills).to match_array( [ claim ])
-      end
-    end
+    # describe 'all_risk_based_bills' do
```
Comment (jsugarman) [inline spec/models/claims/allocation_filters_spec.rb]: remove if not required

**Review by jsugarman — COMMENTED** [review]
hi, looks ok but i don't have enough insight to check whether there are any N+1 queries going on. I'd refactor that method with lots of conditional logic tho. 👍

**Comment by kmahern** [comment]
Hi Joel, think I have made the changes you suggested, Re the N+1 queries issue, I also don't know for sure but I don't think I have changed the underlying way that the filter is working. I have released it to Staging if you want to take a look, I have had a quick test and the performance doesn't appear any different to me.

**Review by colinbruce — APPROVED** [review]
👍

**Pull request merged**